File: server/Simulation/Components/InventoryComponent.cs
using System.Runtime.InteropServices;
using server.ECS;
using server.Enums;

namespace server.Simulation.Components;

[Component(ComponentType = ComponentType.Inventory, NetworkSync = true)]
[StructLayout(LayoutKind.Sequential, Pack = 1)]
public struct InventoryComponent(int storageSpace)
{
    /// <summary>
    /// Tick when this component was last changed, used for network sync.
    /// MUST be first field for raw byte access in ComponentSerializer.
    /// </summary>
    public long ChangedTick = NttWorld.Tick;

    public int TotalCapacity = storageSpace;
    public int Triangles = 0;
    public int Squares = 0;
    public int Pentagons = 0;
}

File: pixiejsClient/src/app/ecs/components/InventoryComponent.ts
import { Component, component, serverField } from "../core/Component";
import { ServerComponentType } from "../../enums/ComponentIds";
import { NTT } from "../core/NTT";

@component(ServerComponentType.Inventory)
export class InventoryComponent extends Component {
  // changedTick is inherited from Component base class
  @serverField(1, "i32") totalCapacity: number;
  @serverField(2, "i32") triangles: number;
  @serverField(3, "i32") squares: number;
  @serverField(4, "i32") pentagons: number;

  constructor(ntt: NTT, totalCapacity: number = 0, triangles: number = 0, squares: number = 0, pentagons: number = 0) {
    super(ntt);
    this.totalCapacity = totalCapacity;
    this.triangles = triangles;
    this.squares = squares;
    this.pentagons = pentagons;
  }
}
